docs/DeveloperGuide.md

```diff
@@ -2,12 +2,343 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+[Original Source](https://github.com/nus-cs2113-AY2425S1/tp)
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Overall Class Diagram
```

Comment:
If the purpose is to show the architecture, perhaps use an architecture diagram instead of a class diagram with a tag showing that "many of the classes are omitted for conciseness".
Then, break down your current class diagram into smaller diagrams per component.

Should there be usage of association labels on dependencies?
Also do include the remaining roles for associations.
Do update all the multiplicities of all classes.
There are some member attributes (e.g. `password` in `Storage` class) that does not have its access modifier.